### ISSUE TITLE
Mine specific coin to its respective wallet when most profitable

### ISSUE THREAD

**Issue by hernandn01** [RainbowMiner/RainbowMiner#106]
Is there already a way to do this? When Coin1, Coin2, and Coin3 are most profitable mine to Wallet1, Wallet2 and Wallet3. For the rest of the coins / algos use regular wallet (Either BTC/LTC/DOGE)

**Comment by RainbowMiner** [comment]
Yes. If you setup an individual coin/currency for a pool that supports coin mining in pools.config.txt (or [C]onfiguration->[P]ools, then that coin/currency will be prefered over the regular wallet.

**Comment by hernandn01** [comment]
There maybe a couple of issues with this approach. I have been able to use ZergPoolCoins with one wallet, once coin and one algo. But when I add a second or third algo, ZergPoolCoins shows as a pool on the RainbowMiner Local Dashboard but with no coins or algos available. 

If I used Config -> Pools - > ZergPoolCoins the first thing that shows is the message No Wallets Defined! although if I type the Coin name it will present the Wallet Address correctly. if under Currency-Symbol that I want to mine, I select the two or three coins, the pool shows with no entries.

![image](https://user-images.githubusercontent.com/41829368/45063118-8ab56b00-b069-11e8-9754-242e337fb80f.png)

![image](https://user-images.githubusercontent.com/41829368/45062936-83da2880-b068-11e8-97ca-5ece67d23418.png)

This is another issue, it let me mine for a while but something is not adding up and the stats gets corrupted. Under YiiMP pool I received this message
![image](https://user-images.githubusercontent.com/41829368/45062576-c1d64d00-b066-11e8-9f29-9e163c6291a1.png)

**Comment by RainbowMiner** [comment]
Oh! I will look into the configuration. It should not warn with no wallets defined. Could you send me your pools.config.txt, please? I will test it locally and mine for you. The Yiimp pool obviously delivers an infinity symbol. I will check for this, so that the error will no longer pop up.
RBM

**Comment by hernandn01** [comment]
[pools.config.txt](https://github.com/RainbowMiner/RainbowMiner/files/2350865/pools.config.txt)

**Comment by RainbowMiner** [comment]
Ok, thank you. I found a bug with multiple coins in the list. Will be fixed asap.
RBM

**Comment by hernandn01** [comment]
trying now version 3.5.8.14 not sure if this was intended, it cleared all my stats but is doing this on every cycle.

![image](https://user-images.githubusercontent.com/41829368/45106456-f8a67480-b0f3-11e8-91ff-c252d8b89d59.png)

**Comment by hernandn01** [comment]
at the end of the cycle
![image](https://user-images.githubusercontent.com/41829368/45106589-458a4b00-b0f4-11e8-9405-081988d089e7.png)

**Comment by RainbowMiner** [comment]
If you have other PHI2 pools, only the current most profitable will be selected. So, if another pool offers more profit for mining PHI2 (for example ZergPool), this one will be used.

**Comment by hernandn01** [comment]
I understand that, but I am looking at All Pools and not Best Pools. I am not filtering by Lux only. I have 3 coins set up on ZergPoolCoins and I only see Lux, if I query the API myself I do see data brought back on the 3 coins.

 "ZergPoolCoins": {
    "BTC": "$Wallet",
    "Worker": "$WorkerName",
    "Penalty": 0.0,
    "User": "$UserName",
    "API_ID": "$API_ID",
    "API_Key": "$API_Key",
    "DataWindow": "estimate_current",
    "LUX": "LfmPRSKxSNxt6WqUzeqVav5RAswnUPin7L",
    "REDN": "RDWKTsMwQKMxEmeDKwAQNj3PfJ6bae4DMB",
    "KREDS": "KEnk53xpLF5xZz6Lq1ogxkd9V9L5ZqHaqP",
    "Algorithm": "phi2,x16s,lyra2v2",
    "ExcludeAlgorithm": "",
    "CoinName": "",
    "ExcludeCoin": "",
    "CoinSymbol": "LUX,REDN,KREDS",
    "ExcludeCoinSymbol": ""
  },

![image](https://user-images.githubusercontent.com/41829368/45167748-053fd100-b1b7-11e8-9ca5-d428767ad178.png)

![image](https://user-images.githubusercontent.com/41829368/45168039-a2026e80-b1b7-11e8-8876-160f4d57cf22.png)

![image](https://user-images.githubusercontent.com/41829368/45168059-ae86c700-b1b7-11e8-833f-e617c5746753.png)

![image](https://user-images.githubusercontent.com/41829368/45168086-bb0b1f80-b1b7-11e8-8b87-cfa913b2cdbe.png)

**Comment by RainbowMiner** [comment]
A miner will be deactivated, if there is no hashrate on a coin or if the transmitted price is older than 1 hour, or if the price is zero. I will check this.  Maybe there is some contrain that can be trimmed.
RBM

**Comment by hernandn01** [comment]
Regardless of profitability mining coins when there is more than one coin per algo are not showing as an option to be mined (REDN,STAKS,KREDS). All wallets per pool are not being shown in the Balance.

![image](https://user-images.githubusercontent.com/41829368/45314015-d21a7c00-b4ed-11e8-846e-c34aebf0fa46.png)

Checked if there were miners for REDN (x16s) and coin was being mined. there were not at the moment for KREDS and STAK (lyra2v2)

![image](https://user-images.githubusercontent.com/41829368/45314063-fc6c3980-b4ed-11e8-889f-1f0fd0cb912c.png)

![image](https://user-images.githubusercontent.com/41829368/45314196-540aa500-b4ee-11e8-9728-9523e1e2253c.png)

**Comment by RainbowMiner** [comment]
I think, I will dig deeper into this, now. Could you send me your pools.config.txt, please?
No, no should not need to declare every single algorithm for the coins. This should be automatically selected.

**Comment by hernandn01** [comment]
[pools.config.txt](https://github.com/RainbowMiner/RainbowMiner/files/2368459/pools.config.txt)

Here it is, I remove all declared algo and just left some exclusions.

**Comment by RainbowMiner** [comment]
The selection contains of the following steps:
1. pool's API must respond and report a hashrate greater than 0 (if it does not respond, the balance list will be empty, too)
2. best pools resp. coins for each algorithm will be selected using the "StablePrice" to compare (not the "LivePrice", to avoid heavy switching). Look into the All Pools in web interface (click on "+" to show all variables).
3. best miners for the best pools will be selected using LivePrice multiply by Hashrate

I did try your config and most of the time BlazePools's, NiceHash's and ZergPool's (Stable-)prices are better than those, reported from the ZergPoolCoins. At most of the time, all coins show up at All Pools > ZergPoolCoins (ORE, REDN fail from time to time). If a coin failed to show up here, it was always because the API reported hashrate has been 0.
After disabling sonoa,lyra2re2,x16s and phi2 at BlazePool, Nicehash and ZergPool , the ZergPoolCoins  showed up in the Best Pools list. 

Since I have changed some things in my running version of RainbowMiner, maybe you wait for the next release.
